duos_ui/lib/screens/login_page.dart

```diff
@@ -28,78 +30,118 @@ class _LoginPageState extends State<LoginPage> {
+                      if (emailValid) {
+                        return null;
+                      }
+                      return 'Enter Valid Email';
+                    }
+                  },
+                  decoration: const InputDecoration(
+                    border: OutlineInputBorder(),
+                    labelText: 'Email',
+                    //hintText:
+                  ),
+                  textInputAction: TextInputAction.next,
                 ),
-                textInputAction: TextInputAction.next,
               ),
-            ),
-            Padding(
-              padding: const EdgeInsets.only(
-                  left: 20.0, right: 20.0, top: 15, bottom: 0),
-              child: TextField(
-                obscureText: true,
-                controller: _passwordController,
-                decoration: const InputDecoration(
-                  border: OutlineInputBorder(),
-                  labelText: 'Password',
-                  hintText: 'Enter password',
+              Padding(
+                padding: const EdgeInsets.only(
+                    left: 20.0, right: 20.0, top: 15, bottom: 0),
+                child: TextFormField(
+                  obscureText: true,
+                  controller: _passwordController,
+                  validator: (val) {
+                    if (val!.isEmpty) {
+                      return 'Empty';
+                    }
+                  },
+                  decoration: const InputDecoration(
+                    border: OutlineInputBorder(),
+                    labelText: 'Password',
+                    hintText: 'Enter password',
+                  ),
                 ),
               ),
-            ),
-            TextButton(
-              onPressed: () => Navigator.of(context).push(
-                MaterialPageRoute(
-                  builder: (context) => const ForgotPasswordPage(),
+              TextButton(
+                onPressed: () => Navigator.of(context).push(
+                  MaterialPageRoute(
+                    builder: (context) => const ForgotPasswordPage(),
+                  ),
+                ),
+                child: const Text(
+                  'Forgot Password?',
+                  style: TextStyle(color: Colors.black, fontSize: 15),
                 ),
               ),
-              child: const Text(
-                'Forgot Password?',
-                style: TextStyle(color: Colors.black, fontSize: 15),
+              Container(
+                height: 50,
+                width: 250,
+                decoration: BoxDecoration(
+                  color: Colors.black,
+                  borderRadius: BorderRadius.circular(20),
+                ),
+                child: TextButton(
+                  onPressed: () {
+                    if (_signinform.currentState!.validate()) {
+                      signIn();
+                      ScaffoldMessenger.of(context).showSnackBar(
```

Comment:
(Optional) Maybe this snackbar is a bit much because we already have the loading spinner with showDialog call in the signin function.